app/src/main/java/com/example/cinemaddict/ui/discover/DiscoverFragment.kt

```diff
@@ -1,8 +1,121 @@
 package com.example.cinemaddict.ui.discover
 
+import android.view.View
+import androidx.fragment.app.viewModels
+import androidx.lifecycle.lifecycleScope
+import androidx.recyclerview.widget.StaggeredGridLayoutManager
+import com.example.cinemaddict.common.paging.MovPagingAdapter
 import com.example.cinemaddict.databinding.FragmentDiscoverBinding
+import com.example.cinemaddict.databinding.ItemDiscoverScreenBinding
+import com.example.cinemaddict.domain.entity.FilmDiscoverData
+import com.example.cinemaddict.ext.toast
 import com.example.cinemaddict.ui.base.BaseUiFragment
+import com.example.cinemaddict.ui.discover.adapter.GenrePagerAdapter
+import com.example.cinemaddict.util.LoadStateListener
+import com.google.android.material.tabs.TabLayoutMediator
 import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.flow.debounce
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.launch
 
 @AndroidEntryPoint
-class DiscoverFragment : BaseUiFragment<FragmentDiscoverBinding>(FragmentDiscoverBinding::inflate)
\ No newline at end of file
+class DiscoverFragment : BaseUiFragment<FragmentDiscoverBinding>(FragmentDiscoverBinding::inflate) {
+
+    private lateinit var adapter: GenrePagerAdapter
+    override val viewModel: DiscoverViewModel by viewModels()
+
+    private val movieAdapter: MovPagingAdapter<FilmDiscoverData, ItemDiscoverScreenBinding> by lazy {
+        MovPagingAdapter(
+            bindingInflater = ItemDiscoverScreenBinding::inflate,
+            onClickListener = ::navigateToDetailScreen
+        )
+    }
+    private val stateListener: LoadStateListener<FilmDiscoverData, ItemDiscoverScreenBinding> by lazy {
+        LoadStateListener(
+            adapter = movieAdapter,
+            toastCallback = ::toast
+        )
+    }
+
+    override fun initViews() {
+        super.initViews()
+        binding.viewModel = viewModel
+        binding.rvList.apply {
+            layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
```

Comment:
```suggestion
            layoutManager = StaggeredGridLayoutManager(SOME_CONSTANT, StaggeredGridLayoutManager.VERTICAL)
```

try to avoid hardcode

Reply:
fixed